### PR TITLE
Add cluster actuator support

### LINKED ISSUE #187: Check how to delete master node
/kind bug
/sig ibmcloud

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

As the cluster controller was running on master node, so we may not able to delete master node based on steps here https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud#cluster-deletion .

@wanghh2000 can you please help check this?

FYI @jichenjc @xunpan @clyang82 @hchenxa

### PR DESCRIPTION
**Pull request by jichenjc**
add simple actuator to avoid empty data so that cluster
can pass validation when doing delete

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #187 

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @xunpan

**Special notes for your reviewer**:

/sig ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go [hunk 1]
```diff
@@ -17,38 +17,70 @@ limitations under the License.
...
 
-// Actuator is responsible for performing cluster reconciliation
-type Actuator struct {
-	clustersGetter client.ClustersGetter
-}
-
-// ActuatorParams holds parameter information for Actuator
-type ActuatorParams struct {
-	ClustersGetter client.ClustersGetter
+type IbmCloudClient struct {
```
Comment (clyang82) [inline pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: `IbmCloudClient` to `IBMCloudClient` for keeping consistence

**Comment by jichenjc** [comment]
with this change, the 
Waiting for Cluster API resources to have statuses... is still there , but much progress made
more PR need to be added but it's related to machine part

@xunpan @gyliu513 please help to review and merge this

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go [hunk 2]
```diff
@@ -17,38 +17,70 @@ limitations under the License.
 package cluster
 
 import (
+	"context"
+	"fmt"
```
Comment (xunpan) [inline pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: Let's add a blank space line between system package and 3rd party package.

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go [hunk 3]
```diff
@@ -17,38 +17,70 @@ limitations under the License.
...
+	params ibmcloud.ActuatorParams
+	client client.Client
 }
 
 // NewActuator creates a new Actuator
-func NewActuator(params ActuatorParams) (*Actuator, error) {
-	return &Actuator{
-		clustersGetter: params.ClustersGetter,
+// NewActuator creates a new Actuator
+func NewActuator(params ibmcloud.ActuatorParams) (*IBMCloudClient, error) {
```
Comment (xunpan) [inline pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: I'm not sure if we need to change the return type name. We try to `NewActuator` but we actually new a `IBMCloudCLient`. 

What about kepping `Actuator` here, it is generated name and without any issue I guess?

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go [hunk 4]
```diff
@@ -17,38 +17,70 @@ limitations under the License.
...
+
+func (a *IBMCloudClient) storeClusterStatus(cluster *clusterv1.Cluster, status *providerv1.IBMCloudClusterProviderStatus) error {
+	ext, err := providerv1.EncodeClusterStatus(status)
+	if err != nil {
+		return fmt.Errorf("failed to update cluster status for cluster %q in namespace %q: %v", cluster.Name, cluster.Namespace, err)
+	}
+	cluster.Status.ProviderStatus = ext
+
+	statusClient := a.params.Client.Status()
+	if err := statusClient.Update(context.Background(), cluster); err != nil {
```
Comment (xunpan) [inline pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: What about `context.TODO()`?

> Background returns a non-nil, empty Context. It is never canceled, has no 
> values, and has no deadline. It is typically used by the main function,  
> initialization, and tests, and as the top-level Context for incoming
> requests.

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go [hunk 5]
```diff
@@ -17,38 +17,70 @@ limitations under the License.
...
-func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
+func (a *IBMCloudClient) Reconcile(cluster *clusterv1.Cluster) error {
+	if cluster == nil {
+		return fmt.Errorf("The cluster is nil, check your cluster configuration")
+	}
+
 	klog.Infof("Reconciling cluster %v.", cluster.Name)
 
-	klog.Infof("Reconciling cluster %v: Not implemented yet.", cluster.Name)
+	// Load provider status.
```
Comment (xunpan) [inline pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: The function name seems already tell the truth.

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go (hunk 3 above)
Comment (jichenjc) [reply pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: it's identical to the machine actuator...

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go (hunk 4 above)
Comment (jichenjc) [reply pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: it's copied from openstack side, I can change that

File: pkg/cloud/ibmcloud/actuators/cluster/actuator.go (hunk 3 above)
Comment (xunpan) [reply pkg/cloud/ibmcloud/actuators/cluster/actuator.go]: @jichenjc 
It is not functional related. Keep it consistent now can be fine.

**Pull request merged**